### DISCUSSION TITLE
python + flask deploy: 500:INTERNAL_SERVER_ERROR

### DISCUSSION THREAD

**Discussion by dooropening** [vercel/community#6837] [Help] [answered]
### Summary

I created the python + flask project based on the [example ](https://github.com/vercel/examples/tree/main/python/flask3)on github. After deploying it to vercel, the prompt 500: INTERNAL_SERVER_ERROR:
This Serverless Function has crashed.
Your connection is working correctly.
Vercel is working correctly.

Checking the running log shows that there is a problem with get/favicon, and it displays:
 LAMBDA_WARNING: Unhandled exception. The most likely cause is an issue in the function code. However, in rare cases, a Lambda runtime update can cause unexpected function behavior. For functions using managed runtimes, runtime updates can be triggered by a function change, or can be applied automatically. To determine if the runtime has been updated, check the runtime version in the INIT_START log entry. If this error correlates with a change in the runtime version, you may be able to mitigate this error by temporarily rolling back to the previous runtime version. For more information, see https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html
[ERROR] Runtime.ImportModuleError: Unable to import module 'vc__handler__python': No module named 'attribute'
Traceback (most recent call last):
But my attribute module is already in the api file directory. structure and word spelling are correct,project runs ok on my local machine,I can't find the problem, please give me some suggestions, thank you.

### Steps to Reproduce

my code: on [github](https://github.com/dooropening/quote-flask)
vercel deploy: [here](https://quote-flask-git-detached-amplescopes-projects.vercel.app/)
screenshot: 
![error_image1](https://github.com/vercel/community/assets/97821818/4b222cd8-58bd-48d4-b7b5-b05b53e46d56)

![error_image](https://github.com/vercel/community/assets/97821818/f7001378-4712-4f85-b729-14945365fa56)

**Comment by dooropening** [comment, marked as answer]
I solved this problem myself. Since there was a problem with importing self-built modules, I wrote all the modules in a py file. There is only one py file in the entire site, index.py. Now the site is successfully deployed on vercel and the 500 error is resolved.

**Reply by dooropening** [reply]
Later, I studied it carefully and found the real problem. It was that the function import used a relative path and an absolute path was required.

**Reply by Rashi2608** [reply]
i am facing path problems

![image](https://github.com/vercel/community/assets/138390220/29225562-0377-47b7-ab39-2fa1761bce3d)
structure looks like this  i need path for "artifacts/model.pkl"

**Reply by Akhil-Raj** [reply]
Hi @dooropening, changing relative imports to absolute worked for me too! Thanks!

I used this to convert relative to absolute imports : https://realpython.com/absolute-vs-relative-python-imports/

[a comment by an automated account is not shown]
